### PR TITLE
Make include_global_phase argument of assert_decompose_is_consistent_with_unitary default to True

### PR DESCRIPTION
**Pull request by smitsanghavi**
And fix broken tests. #1161

### CODE REVIEW

File: cirq/testing/consistent_protocols.py [hunk 1]
```diff
@@ -52,6 +55,7 @@ def assert_eigengate_implements_consistent_protocols(
             0, 1, -1, 0.5, 0.25, -0.5, 0.1, value.Symbol('s')),
         global_shifts: Sequence[float] = (0, 0.5, -0.5, 0.1),
         qubit_count: Optional[int] = None,
+        is_global_phase_sensitive: bool=False,
```
Comment (kevinsung) [inline cirq/testing/consistent_protocols.py]: I think we agreed in #1161 that global phase should always be preserved, so you should not be adding this flag.

Comment (Strilanc) [reply cirq/testing/consistent_protocols.py]: Long term we want global phase to be preserved, but short term it makes sense to have this.

Rename to `ignoring_global_phase` defaulting to false. We want to be able to find tests opting into global phase insensitivity and update them as we fix individual issues.

File: cirq/testing/consistent_decomposition.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 def assert_decompose_is_consistent_with_unitary(
         val: Any,
-        including_global_phase: bool=False):
+        including_global_phase: bool=True):
```
Comment (Strilanc) [inline cirq/testing/consistent_decomposition.py]: While we're switching the default value, let's rename this to `ignoring_global_phase` (inverting its meaning).

File: cirq/ops/three_qubit_gates_test.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 )
 def test_eigen_gates_consistent_protocols(eigen_gate_type):
     cirq.testing.assert_eigengate_implements_consistent_protocols(
-            eigen_gate_type)
+            eigen_gate_type, is_global_phase_sensitive = True)
```
Comment (Strilanc) [inline cirq/ops/three_qubit_gates_test.py]: Isn't this the default now? Don't specify it.

File: cirq/ops/phased_x_gate_test.py [hunk 4]
```diff
@@ -28,7 +28,14 @@ def test_phased_x_consistent_protocols(phase_exponent):
     cirq.testing.assert_implements_consistent_protocols(
             cirq.PhasedXPowGate(phase_exponent=phase_exponent,
                                 exponent=1.0,
-                                global_shift=0.1)
+                                global_shift=0),
+            is_global_phase_sensitive = False
+    )
+    cirq.testing.assert_implements_consistent_protocols(
```
Comment (Strilanc) [inline cirq/ops/phased_x_gate_test.py]: Why the duplicated test? The global phase sensitive version passing implies the global phase insensitive version will pass. Undo changes to this file.

File: cirq/ops/common_gates_test.py [hunk 5]
```diff
@@ -26,19 +26,20 @@
...
-@pytest.mark.parametrize('eigen_gate_type', [
-    cirq.CNotPowGate,
-    cirq.CZPowGate,
-    cirq.HPowGate,
-    cirq.ISwapPowGate,
-    cirq.SwapPowGate,
-    cirq.XPowGate,
-    cirq.YPowGate,
-    cirq.ZPowGate,
+@pytest.mark.parametrize('eigen_gate_and_phase_sensitivity', [
```
Comment (Strilanc) [inline cirq/ops/common_gates_test.py]: Split this into two different tests, one for the false case and one for true.

Comment (Strilanc) [inline cirq/ops/common_gates_test.py]: (You use commas to separate arguments in the spec string.)

File: cirq/testing/consistent_protocols.py (hunk 1 above)
Comment (smitsanghavi) [reply cirq/testing/consistent_protocols.py]: Renamed.

File: cirq/testing/consistent_decomposition.py (hunk 2 above)
Comment (smitsanghavi) [reply cirq/testing/consistent_decomposition.py]: Renamed, hence inverting meaning. So didn't change the default value.

File: cirq/ops/three_qubit_gates_test.py (hunk 3 above)
Comment (smitsanghavi) [reply cirq/ops/three_qubit_gates_test.py]: The default is False.

File: cirq/ops/phased_x_gate_test.py (hunk 4 above)
Comment (smitsanghavi) [reply cirq/ops/phased_x_gate_test.py]: It's not duplicated. The phase insensitive version passes only if global_shift==0. 
Added the insensitive test to verify that global phase doesn't change if the arg is 0.

File: cirq/ops/phased_x_gate_test.py [hunk 6]
```diff
@@ -25,10 +25,18 @@
     ]
 )
 def test_phased_x_consistent_protocols(phase_exponent):
+    # If there is no global_shift, the gate is global phase insensitive.
     cirq.testing.assert_implements_consistent_protocols(
             cirq.PhasedXPowGate(phase_exponent=phase_exponent,
                                 exponent=1.0,
-                                global_shift=0.1)
+                                global_shift=0),
+            ignoring_global_phase = False
```
Comment (Strilanc) [inline cirq/ops/phased_x_gate_test.py]: Don't bother to say you're using the default value.

File: cirq/ops/phased_x_gate_test.py [hunk 7]
```diff
@@ -25,10 +25,18 @@
...
+    )
+    cirq.testing.assert_implements_consistent_protocols(
+            cirq.PhasedXPowGate(phase_exponent=phase_exponent,
+                                exponent=1.0,
+                                global_shift=0.1),
+            ignoring_global_phase = True
```
Comment (Strilanc) [inline cirq/ops/phased_x_gate_test.py]: Nit: no spaces around the `=`.

**Pull request merged**